Tests/test_file_avif.py

```diff
@@ -0,0 +1,830 @@
+        """
+        Convert an animated GIF to animated AVIF, then compare the frame
+        count, and first and last frames to ensure they're visually similar.
+        """
+
+        with Image.open("Tests/images/avif/star.gif") as orig:
+            assert orig.n_frames > 1
+
+            temp_file = str(tmp_path / "temp.avif")
+            orig.save(temp_file, save_all=True)
+            with Image.open(temp_file) as im:
+                assert im.n_frames == orig.n_frames
+
+                # Compare first and second-to-last frames to the original animated GIF
+                orig.load()
+                im.load()
+                assert_image_similar(im.convert("RGB"), orig.convert("RGB"), 25.0)
+                orig.seek(orig.n_frames - 2)
+                im.seek(im.n_frames - 2)
+                orig.load()
+                im.load()
+                assert_image_similar(im.convert("RGB"), orig.convert("RGB"), 25.0)
+
+    def test_write_animation_RGB(self, tmp_path: Path) -> None:
+        """
+        Write an animated AVIF from RGB frames, and ensure the frames
+        are visually similar to the originals.
+        """
+
+        def check(temp_file: str) -> None:
+            with Image.open(temp_file) as im:
+                assert im.n_frames == 4
+
+                # Compare first frame to original
+                im.load()
+                assert_image_similar(im, frame1.convert("RGBA"), 25.0)
+
+                # Compare second frame to original
+                im.seek(1)
+                im.load()
+                assert_image_similar(im, frame2.convert("RGBA"), 25.0)
+
+        with self.star_frames() as frames:
+            frame1 = frames[0]
+            frame2 = frames[1]
+            temp_file1 = str(tmp_path / "temp.avif")
+            frames[0].copy().save(temp_file1, save_all=True, append_images=frames[1:])
+            check(temp_file1)
+
+            # Tests appending using a generator
+            def imGenerator(
+                ims: list[ImageFile.ImageFile],
+            ) -> Generator[ImageFile.ImageFile, None, None]:
+                yield from ims
+
+            temp_file2 = str(tmp_path / "temp_generator.avif")
+            frames[0].copy().save(
+                temp_file2,
+                save_all=True,
+                append_images=imGenerator(frames[1:]),
+            )
+            check(temp_file2)
+
+    def test_sequence_dimension_mismatch_check(self, tmp_path: Path) -> None:
+        temp_file = str(tmp_path / "temp.avif")
+        frame1 = Image.new("RGB", (100, 100))
+        frame2 = Image.new("RGB", (150, 150))
+        with pytest.raises(ValueError):
+            frame1.save(temp_file, save_all=True, append_images=[frame2], duration=100)
+
+    def test_heif_raises_unidentified_image_error(self) -> None:
+        with pytest.raises(UnidentifiedImageError):
+            with Image.open("Tests/images/avif/rgba10.heif"):
+                pass
```

Comment:
What is the relevance of this test?

Reply:
It is it ended to test this bit of logic:

https://github.com/python-pillow/Pillow/blob/ee3c46af4259858d2b4b018aa68e5b7d424d794e/src/PIL/AvifImagePlugin.py#L36-L44

If the ftyp box is `avif` or `avis` then this is definitely an AVIF image, and so it is sensible to return a string from the accept function to trigger an error. But `mif1` and `msf1` are also valid values for AVIF, but also for HEIF. The initial bytes passed to the accept function is not long enough to determine which format it is. In order to be interoperable with pyheif I cannot raise an error in the accept, and must instead raise a SyntaxError in `_open` if AVIF is not compiled. `_open` would only be called if there isn't another PIL image plugin that accepts HEIF files.